File: Baekjoon/11053_가장 긴 증가하는 부분수열 길이/B_11053.py
N = int(input())
numbers = list(map(int, input().split()))
dp = [0] * N
dp[0] = 1
if N > 1:
    for i in range(1, N):
        max_cnt = 1
        for j in range(i - 1, -1, -1):
            if numbers[i] > numbers[j]:
                dp[i] = max(max_cnt, dp[j] + 1)
                max_cnt = dp[i]

        if max_cnt == 1:
            dp[i] = 1

print(max(dp))

File: baekjoonn/11053_가장 긴 증가하는 부분수열 길이/B_11053.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const N = Number(input[0])
const sequence = input[1].split(' ').map(v => Number(v))

let lis = new Array(N).fill(1)

for (let i = N - 2; i > -1; i--) {
    const target = sequence[i]
    for (let j = N - 1; j > i; j--) {
        const candidate = sequence[j]
        if (target < candidate) {
            lis[i] = Math.max(lis[j] + 1, lis[i])
        }
    }
}

console.log(Math.max(...lis))